packages/synapse-core/src/warm-storage/data-sets.ts

```diff
@@ -82,7 +82,9 @@ export async function getDataSets(client: Client<Transport, Chain>, options: Get
     })
 
     // getProviderWithProduct returns {providerId, providerInfo, product, productCapabilityValues}
-    const pdpCaps = decodeCapabilities(pdpOffering.product.capabilityKeys, pdpOffering.productCapabilityValues)
+    const pdpCaps = decodePDPCapabilities(
+      capabilitiesListToObject(pdpOffering.product.capabilityKeys, pdpOffering.productCapabilityValues)
+    )
```

Comment:
instead of this can we just call decodePDPCapabilities with keys and values and call capabilitiesListToObject inside?

Reply:
yes that is the part I understood. I was asking about the second part.

Reply:
The reason not to do this is because in synapse-sdk we need the capabilities object too.